src/index.ts

```diff
@@ -46,37 +61,83 @@ const concatenatePageBundles = ({
     return outFile
   })
 
+const getMaxSize = (
+  pageBundleName: string,
+  fallbackSize: string,
+  oldConfiguration?: BundleSizeConfig
+): string => {
+  if (!oldConfiguration) return fallbackSize
+
+  const oldPageConfig = oldConfiguration.files.find(
+    (page) => page.path === pageBundleName
+  )
+  return oldPageConfig && oldPageConfig.maxSize
+    ? oldPageConfig.maxSize
+    : fallbackSize
+}
+
 const generateBundleSizeConfig = ({
   pageBundles,
   maxSize,
+  previousConfiguration,
 }: {
   pageBundles: string[]
   maxSize: string
-}): string => {
-  const config = {
-    files: pageBundles.map((pageBundle) => ({
-      path: pageBundle,
-      maxSize: maxSize,
+  previousConfiguration?: BundleSizeConfig
+}): BundleSizeConfig => {
+  return {
+    files: pageBundles.map((pageBundleName) => ({
+      path: pageBundleName,
+      maxSize: getMaxSize(pageBundleName, maxSize, previousConfiguration),
     })),
   }
-  return JSON.stringify(config)
+}
+
+const extractArgs = (args) => {
+  const parsedArgs = parse(args) as unknown as Args
+  const maxSize = parsedArgs.maxSize || "200 kB"
+  const buildDir = parsedArgs.buildDir || ".next"
+  const delta = (parsedArgs.delta || 2) * 1024
+  const targetSize = (parsedArgs.targetSize || 130) * 1024
+
+  return {
+    maxSize,
+    buildDir,
+    delta,
+    previousConfigFileName: parsedArgs.previousConfigFileName,
+    targetSize,
+  }
 }
 
 export default function run(args) {
   try {
-    const parsedArgs = parse(args) as unknown as Args
-    const maxSize = parsedArgs.maxSize || "200 kB"
-    const buildDir = parsedArgs.buildDir || ".next"
-
+    const { maxSize, buildDir, delta, previousConfigFileName, targetSize } =
+      extractArgs(args)
     const manifestFile = path.join(buildDir, "build-manifest.json")
     const manifest = JSON.parse(fs.readFileSync(manifestFile).toString())
 
     const pageBundles = concatenatePageBundles({ buildDir, manifest })
-    const config = generateBundleSizeConfig({ pageBundles, maxSize })
+    const previousConfiguration = getPreviousConfig(
+      buildDir,
+      previousConfigFileName
+    )
+    const config = generateBundleSizeConfig({
+      pageBundles,
+      maxSize,
+      previousConfiguration,
+    })
     const configFile = path.join(buildDir, "next-page-bundlesize.config.json")
-    fs.writeFileSync(configFile, config)
+    fs.writeFileSync(configFile, JSON.stringify(config))
```

Comment:
If you want pretty formatted output I'd go for:
```suggestion
    fs.writeFileSync(configFile, JSON.stringify(config, null, 2))
```

Reply:
thanks, didn't know that!